xmtp_content_types/src/transaction_reference.rs

```diff
@@ -55,14 +55,32 @@ impl TransactionReferenceCodec {
     }
 }
 
+/// Custom deserializer for network_id that can handle both string and number
+fn deserialize_network_id<'de, D>(deserializer: D) -> Result<String, D::Error>
```

Comment:
This got caught by the interop tests. Turns out the network ID can be a number or a string in the content type.